view/src/components/foodbanks.js

```diff
@@ -230,8 +256,425 @@ class Foodbank extends Component {
+          value.map((option, index) => (
+            <Chip label={option} {...getTagProps({ index })} />
+          ))
+        }
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            variant="outlined"
+            label="Food Bank Tags"
+            placeholder="Type or Select a Food Bank Tag"
+            InputProps={{
+              ...params.InputProps,
+              startAdornment: (
+                <>
+                  <InputAdornment position="start">
+                    <SearchIcon />
+                  </InputAdornment>
+                  {params.InputProps.startAdornment}
+                </>
+              ),
+            }}
+          />
+        )}
+      />
+    );
+  };
+
+  /** Updates filteredData with a new array; used with filters.js queries */
+  updateCards = (newValues) => {
+    if (newValues.length === 0) {
+      return;
+    }
+    this.setState({
+      filteredData: [...newValues],
+    });
+  };
+
+  /** Reset filteredData to the original page data, upon clicking reset */
+  resetCards = () => {
+    this.setState(
+      {
+        filteredData: [...this.state.foodbanks],
+        nameQuery: "",
+        locationQuery: "",
+        tagsQuery: [],
+      },
+      this.populateAllTags
+    );
+  };
+
+  /** End functions for sorting menu on top of the page */
+  /** Renders filtered produce results into Material-UI cards */
+  handleResultsRender = () => {
+    const { classes } = this.props;
+    const { filteredData } = this.state;
+
+    return (
+      <div>
+        <Grid container spacing={2} alignItem="center">
+          {this.state.filteredData.map((foodbank) => (
+            <Grid item xs={12}>
+              <Card
+                className={classes.root}
+                raised={foodbank.foodbankId === this.state.selectedCard}
+                variant={
+                  foodbank.foodbankId === this.state.selectedCard
+                    ? "elevation"
+                    : "outlined"
+                }
+              >
+                <CardContent>
+                  <Typography variant="h5" component="h2">
+                    {foodbank.foodbankName}
+                  </Typography>
+                  {foodbank.foodbankTags.map((tag) => (
+                    <Chip className={classes.chip} label={tag} size="small" />
+                  ))}
+                  <Box
+                    display="flex"
+                    flexDirection="row"
+                    flexWrap="wrap"
+                    padding={0}
+                    margin={0}
+                  >
+                    <Box padding={3}>
+                      <Typography
+                        className={classes.position}
+                        color="textSecondary"
+                      >
+                        Details:
+                      </Typography>
+                      <Typography
+                        variant="body2"
+                        component="p"
```

Comment:
this is p as in `<p>`, yea? 

grumble grumble React/MaterialUI

Reply:
hahaha, yes indeed